rpcs3/Emu/RSX/Common/GLSLCommon.h

```diff
@@ -472,10 +472,31 @@ namespace glsl
 			OS << "	return pow((cs + 0.055) / 1.055, 2.4);\n";
 			OS << "}\n\n";
 
+#ifdef __APPLE__
+			OS << "vec4 remap_vector(vec4 rgba, uint remap_bits)\n";
+			OS << "{\n";
+			OS << "	uint a = (remap_bits & 0x7);\n";
+			OS << "	uint r = (remap_bits >> 3) & 0x7;\n";
+			OS << "	uint g = (remap_bits >> 6) & 0x7;\n";
+			OS << "	uint b = (remap_bits >> 9) & 0x7;\n";
+			OS << "\n";
+			OS << "	vec4 result;\n";
+			OS << "	result.a = (a > 1)? rgba[a - 2] : float(a);\n";
```

Comment:
Would be worth checking if multiplication by a 4x4 matrix ends up cheaper, or just rewriting it straight away like this. Pure arithmetic may be easier on compiler/HW than a bunch of conditional vector indexing operations

Reply:
I can't think of any simple arithmetic to extract components dynamically off the top of my head without introducing more complexity. The challenge here is the component indexing; if it was possible to do that in vector then this could be simplified into 2 or 3 instructions.